### ISSUE TITLE
NPE during legacy migration to 9.0.0

### ISSUE THREAD

**Issue by h3nrique** [keycloak/keycloak#15020]
### Describe the bug

When try migrate version to 9.0.0 throws NPE if the keycloak administrator had removed client 'account'.

### Version

4.6.0

### Expected behavior

Ignore existence of this client and continue database migration.

### Actual behavior

Stops migration with NPE and not rollback all database modification.

```
10:15:29,630 FATAL [org.keycloak.services] (ServerService Thread Pool -- 61) Error during startup: java.lang.NullPointerException
        at org.keycloak.keycloak-server-spi-private@18.0.0//org.keycloak.migration.migrators.MigrateTo9_0_0.addAccountConsoleClient(MigrateTo9_0_0.java:95)
        at org.keycloak.keycloak-server-spi-private@18.0.0//org.keycloak.migration.migrators.MigrateTo9_0_0.migrateRealmCommon(MigrateTo9_0_0.java:59)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
```

#### Steps
1- Create a environment with version before 9.0.0 (In my case 4.6.0)
2- Create a new realm and remove 'account' client.
3 - Upgrade Keycloak pointing to same database  (I'm tried upgrade to 18.0.0)

#### compose.yaml example
```
version: "3"

services:
  postgres-keycloak:
    container_name: postgres-keycloak
    image: postgres:14
    volumes:
      - postgres-keycloak-data:/var/lib/postgresql/data
    environment:
      TZ: America/Fortaleza
      POSTGRES_DB: keycloak
      PGDATABASE: keycloak
      POSTGRES_USER: test
      PGUSER: test
      POSTGRES_PASSWORD: test4120
      PGPASSWORD: test4120

  keycloak460:
    container_name: keycloak460
    image: jboss/keycloak:4.6.0.Final
    ports:
      - "8085:8080"
    environment:
      TZ: America/Fortaleza
      DB_VENDOR: postgres
      DB_ADDR: postgres-keycloak
      DB_PORT: 5432
      DB_DATABASE: keycloak
      DB_USER: test
      DB_PASSWORD: test4120
      KEYCLOAK_USER: test
      KEYCLOAK_PASSWORD: test4120
      JGROUPS_DISCOVERY_PROTOCOL: dns.DNS_PING
      JGROUPS_DISCOVERY_PROPERTIES: dns_query=localhost

  # keycloak1800: 
  #   container_name: keycloak1800
  #   image: quay.io/keycloak/keycloak:18.0.0-legacy
  #   ports:
  #     - "8085:8080"
  #   environment:
  #     TZ: America/Fortaleza
  #     DB_VENDOR: postgres
  #     DB_ADDR: postgres-keycloak
  #     DB_PORT: 5432
  #     DB_DATABASE: keycloak
  #     DB_USER: test
  #     DB_PASSWORD: test4120
  #     KEYCLOAK_USER: test
  #     KEYCLOAK_PASSWORD: test4120
  #     JGROUPS_DISCOVERY_PROTOCOL: dns.DNS_PING
  #     JGROUPS_DISCOVERY_PROPERTIES: dns_query=localhost

  pgadmin-keycloak:
    container_name: pgadmin-keycloak
    image: dpage/pgadmin4:latest
    environment:
      TZ: America/Fortaleza
      PGADMIN_DEFAULT_EMAIL: admin@example.com
      PGADMIN_DEFAULT_PASSWORD: test4120
    ports:
      - 8089:80
    depends_on:
      - postgres-keycloak

volumes:
  postgres-keycloak-data:
```

**Comment by stianst** [comment]
The account client is an internal client that you are not supposed to deleted, and it should not have been possible to delete it really, but we never got around to protecting that. I would suggest just recreating the client then you should be able to migrate just fine.

**Comment by h3nrique** [comment]
So, the migration step should recreate the client 'account' as already do with client 'account-console', right? I'll do the commit here.

**Comment by abstractj** [comment]
@h3nrique I'm slightly confused, which version of Keycloak are you running? The version mention 4.6.0, but at the same time the title says 9.0.0.

**Comment by h3nrique** [comment]
I'm running 4.6.0, but I trying migrate to 18.0.0.

When I run keycloak 18.0.0 pointing to database with 4.6.0 structure, starts the migration steps and throws exception at MigrateTo9_0_0 class.

As the Stian comment, I understand that the 'account' client is required for migration, but the class MigrateTo9_0_0  does not check if exists before do the client change, however in this same class exists a validation for 'account-console' client before a similar change. In the case of 'account-console' client, exists a method that recriate it.

**Comment by stianst** [comment]
I'm not saying the client should be created by the migration. I'm saying you shouldn't have deleted it, so recreate it manually on your end before migrating and it should all work.

**Comment by h3nrique** [comment]
Hi Stian,

When I recreate the client manually the migration worked, however after I removed some itens in COMPOSITE_ROLE and KEYCLOAK_ROLE tables because when occurs the error that originate this issue, a rollback was not fully made in this tables (I not checked others tables about this rollback).

The interpretation that I made about your first message it's because already exists steps in migration to automatically recreate the 'account-console' client if not exists, so, I believe that the same is needed para o 'account'.

The idea is use the same recreation steps for 'account-console' and 'account', because both are core required for Keycloak, since it's not possible protect it for deletion.
